controller/src/main/java/io/pravega/controller/store/InMemoryScope.java

```diff
@@ -41,7 +41,7 @@
     private HashMap<String, Integer> streamsPositionMap;
 
     @GuardedBy("$lock")
-    private TreeMap<String, InMemoryKVTable> kvTablesMap;
+    private final TreeMap<String, InMemoryKVTable> kvTablesMap =  new TreeMap<String, InMemoryKVTable>();
```

Comment:
this can be reverted except the `final` keyword.

Reply:
Why?

Reply:
no i mean init in constructor

Reply:
Construction initialization is good for usecases where the values used in building the object need to be passed to the constructor. Initializing at the time of declaration also takes away the need to synchronize the constructor and I don't see any downsides with initializing it this way.

Reply:
no downside, just consistency as other objects are initialized in the constructor.

Also constructor is implicitly synchronized and you dont need to synchronize a constructor.

Reply:
Ok.